src/measure_theory/lp_space.lean

```diff
@@ -1218,4 +1217,409 @@ by { rw eq, exact continuous_pos_part.comp continuous_neg }
+  (h : ∀ n, ∫⁻ a, (∑ i in finset.range (n + 1), nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ
+    ≤ (∑' i, B i) ^ p) :
+  (∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ) ^ (1/p) ≤ ∑' i, B i :=
+begin
+  have hp_pos : 0 < p := zero_lt_one.trans_le hp1,
+  suffices h_pow : ∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ ≤ (∑' i, B i) ^ p,
+    by rwa [←@ennreal.le_rpow_one_div_iff _ _ (1/p) (by simp [hp_pos]), one_div_one_div],
+  have h_tsum_1 : ∀ g : ℕ → ℝ≥0∞,
+      ∑' i, g i = at_top.liminf (λ n, ∑ i in finset.range (n + 1), g i),
+    by { intro g, rw [ennreal.tsum_eq_liminf_sum_nat, ← liminf_nat_add _ 1], },
+  simp_rw h_tsum_1 _,
+  rw ← h_tsum_1,
+  have h_liminf_pow : ∫⁻ a, at_top.liminf (λ n, ∑ i in finset.range (n + 1),
+      (nnnorm (f (i + 1) a - f i a)))^p ∂μ
+    = ∫⁻ a, at_top.liminf (λ n, (∑ i in finset.range (n + 1), (nnnorm (f (i + 1) a - f i a)))^p) ∂μ,
+  { refine lintegral_congr (λ x, _),
+    have h_rpow_mono := ennreal.rpow_left_strict_mono_of_pos (zero_lt_one.trans_le hp1),
+    have h_rpow_surj := (ennreal.rpow_left_bijective hp_pos.ne.symm).2,
+    refine (h_rpow_mono.order_iso_of_surjective _ h_rpow_surj).liminf_apply _ _ _ _,
+    all_goals { is_bounded_default }, },
+  rw h_liminf_pow,
+  refine (lintegral_liminf_le' _).trans _,
+  { exact λ n, ae_measurable.ennreal_rpow_const (finset.ae_measurable_sum (finset.range (n+1))
+      (λ i, ((hf (i+1)).sub (hf i)).nnnorm.ennreal_coe)), },
+  { exact liminf_le_of_frequently_le' (frequently_of_forall h), },
+end
+
+private lemma tsum_nnnorm_sub_ae_lt_top
+  {f : ℕ → α → E} (hf : ∀ n, ae_measurable (f n) μ) {p : ℝ} (hp1 : 1 ≤ p) {B : ℕ → ℝ≥0∞}
+  (hB : ∑' i, B i < ∞)
+  (h : (∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ) ^ (1/p) ≤ ∑' i, B i) :
+  ∀ᵐ x ∂μ, (∑' i, nnnorm (f (i + 1) x - f i x) : ℝ≥0∞) < ∞ :=
+begin
+  have hp_pos : 0 < p := zero_lt_one.trans_le hp1,
+  have h_integral : ∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ < ∞,
+  { have h_tsum_lt_top : (∑' i, B i) ^ p < ∞,
+      from ennreal.rpow_lt_top_of_nonneg hp_pos.le (lt_top_iff_ne_top.mp hB),
+    refine lt_of_le_of_lt _ h_tsum_lt_top,
+    rwa [←@ennreal.le_rpow_one_div_iff _ _ (1/p) (by simp [hp_pos]), one_div_one_div] at h, },
+  have rpow_ae_lt_top : ∀ᵐ x ∂μ, (∑' i, nnnorm (f (i + 1) x - f i x) : ℝ≥0∞)^p < ∞,
+  { refine ae_lt_top' (ae_measurable.ennreal_rpow_const _) h_integral,
+    exact ae_measurable.ennreal_tsum (λ n, ((hf (n+1)).sub (hf n)).nnnorm.ennreal_coe), },
+  refine rpow_ae_lt_top.mono (λ x hx, _),
+  rwa [←ennreal.lt_rpow_one_div_iff hp_pos,
+    @ennreal.top_rpow_of_pos (1/p) (by simp [hp_pos])] at hx,
+end
+
+lemma ae_tendsto_of_cauchy_snorm' [complete_space E] {f : ℕ → α → E} {p : ℝ}
+  (hf : ∀ n, ae_measurable (f n) μ) (hp1 : 1 ≤ p) {B : ℕ → ℝ≥0∞} (hB : ∑' i, B i < ∞)
+  (h_cau : ∀ (N n m : ℕ), N ≤ n → N ≤ m → snorm' (f n - f m) p μ < B N) :
+  ∀ᵐ x ∂μ, ∃ l : E, at_top.tendsto (λ n, f n x) (𝓝 l) :=
```

Comment:
This lemma also holds for `p = top` (this is the main step of the proof of `cauchy_complete_ℒp_top`). Could you then prove it uniformly for all `p \ge 1` and finish the proof from there? (I am trying to get a proof that separates between different values of `p` only what really has to be separated). This will make it possible to prove uniformly over `p` lemmas such as: a converging sequence in `L^p` admits an ae-convergent subsequence.

Reply:
Done. Thanks for suggesting those changes. The proof is looking prettier after each iteration.

Reply:
You still have `cauchy_complete_ℒp_of_ne_top` and `cauchy_complete_ℒp_top`, but normally it should be possible to merge them now (their proofs should be exactly the same!)

Reply:
Right, I'll change that. I think that I stared at that file so much in the last two days that I can't see anything.

Reply:
There is now only one `cauchy_complete_ℒp`